assertj-error-prone/src/test/java/com/palantir/assertj/errorprone/AssertjAssertThatThrownByTest.java

```diff
@@ -0,0 +1,263 @@
+                        "  void foo() {",
+                        "    try {",
+                        "      System.out.println(\"1\");",
+                        "      System.out.println(\"2\");",
+                        "      System.out.println(\"3\");",
+                        "      fail(\"fail\");",
+                        "    } catch (RuntimeException expected) {}",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "MyClass.java",
+                        "import static org.assertj.core.api.Assertions.assertThatThrownBy;",
+                        "import static org.junit.Assert.fail;",
+                        "",
+                        "import org.junit.jupiter.api.Test;",
+                        "",
+                        "class MyClass {",
+                        "  @Test",
+                        "  void foo() {",
+                        "    assertThatThrownBy(() -> {",
+                        "      System.out.println(\"1\");",
+                        "      System.out.println(\"2\");",
+                        "      System.out.println(\"3\");",
+                        "    }).isInstanceOf(RuntimeException.class);",
+                        "  }",
+                        "}")
+                .doTest(BugCheckerRefactoringTestHelper.TestMode.TEXT_MATCH);
+    }
+
+    @Test
+    public void fix_while_preserving_fail_message() {
+        RefactoringValidator.of(new AssertjAssertThatThrownBy(), getClass())
+                .addInputLines(
+                        "MyClass.java",
+                        "import static org.junit.Assert.fail;",
+                        "",
+                        "import org.junit.jupiter.api.Test;",
+                        "",
+                        "class MyClass {",
+                        "  @Test",
+                        "  void foo() {",
+                        "    try {",
+                        "      System.out.println();",
+                        "      fail(\"My error message.\");",
+                        "    } catch (RuntimeException expected) {}",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "MyClass.java",
+                        "import static org.assertj.core.api.Assertions.assertThatThrownBy;",
+                        "import static org.junit.Assert.fail;",
+                        "",
+                        "import org.junit.jupiter.api.Test;",
+                        "",
+                        "class MyClass {",
+                        "  @Test",
+                        "  void foo() {",
+                        "    assertThatThrownBy(() -> {",
+                        "      System.out.println();",
+                        "    })",
+                        "    .isInstanceOf(RuntimeException.class)",
+                        "    .describedAs(\"My error message.\");",
```

Comment:
I don't think this actually works (and will fail another error-prone check for ignored return values). The describedAs block must occur prior to checks, which fail immediately when they're executed.

Reply:
Had problems reproducing this. It might only apply for a newer version of AssertJ? Anyway, I switched the order of the `.describedAs()` and `.isInstanceOf()` calls so that `describedAs()` comes first.